enhancements/metering/slim-install.md

```diff
@@ -0,0 +1,210 @@
+- In the case a user wishes to install Metering in a disconnected, air-gapped, or simply on an environment that does not have access to cloud storage, Metering requires the usage of a StorageClass or PersistentVolume with ReadWriteMany (RWX) access modes.
+- At a minimum, Metering requires the user to define a storage configuration in the `MeteringConfig` custom resource and that leaves the door open to potential misconfigurations or difficulties setting up the pre-requisites for Metering usage.
+
+### Goals
+
+- Reduce the complexity of the Metering stack.
+- Reduce the overall CPU and memory footprint of the Metering stack.
+- New Metering installations default to enabling the slim install configuration.
+- A potential migration path is outlined.
+
+### Non-Goals
+
+- Initial support for cloud object storage.
+- Maintaining support for AWS billing integration.
+- Metering is delivered as a CVO-based Operator.
+
+## Proposal
+
+Currently, and by default, Metering periodically imports and stores a small subset of the available Prometheus metrics in Hive long term, using the Prometheus query_range API. Historically, Presto did not have the ability to directly query Prometheus, but it does have the ability to integrate with Hive datasets. This results in the following workaround, where tables are needed to be first created in Hive, and then newer metrics would be inserted by Presto into existing tables in Hive using the `INSERT INTO ... VALUES(...)` operation. It's also worth noting that type of operation is particularly expensive for Presto in comparison to Postgres, which is designed as a more transactional-based system (OLTP).
+
+Due to the nature of periodic importing, most of the components are sitting idle, especially in the case of Presto which is still utilizing resources despite no queries being processed or evaluated.
+
+With the introduction of Postgres, and the removal of Hive and Presto components, we can continue to utilize the existing Prometheus metrics importer implementation, adapting to the native Postgres data types, and the result is a more streamlined approach to data flow and an overall less complex stack.
+
+This implementation would allow Metering to continue using the core, existing CustomResourceDefinitions (CRDs) already exposed in previous releases:
+
+- `ReportDataSource`: exposes an existing table in Postgresql that can be used for reporting needs.
+- `ReportQuery`: defines a SQL query or view that references a `ReportDataSource` table, or a table created by a `Report` object.
+- `Report`: evaluates a templated SQL query defined in an existing `ReportQuery` custom resource and stores those query results in a Hive (now Postgres) table.
+
+At a high-level, the following must be done:
+
+- Create a `PostgresTable` CRD, which is an abstraction that represents an individual table in Postgres.
```

Comment:
not clear to me why we need this intermediary resource. What generates these CRs and what consumes them?  And why can't the thing that's generating them(or consuming them) just do the work of creating/updating the postgres tables directly?

Reply:
I think I generally fall in line that anything configurable, which could be the case here, should be defined as a CRD, but this is essentially just building off the existing set of metering-related CRDs, like `HiveTable` and `PrestoTable`. Those CRDs are more-or-less internal to the reporting-operator, but it's still possible for the user to define them explicitly and there are cases in the current version of the operator where that would make sense.

> What generates these CRs and what consumes them?

When the reporting-operator is watching and processing new/existing `ReportDataSource` and `Report` custom resources, if no database table has been created (e.g. check the status field of those resources) then create a `PostgresTable` object with the owner GVK being the `ReportDataSource`/`Report` we're currently processing. So the `PostgresTable` object would be responsible for making the call to create a database table, but the `ReportDataSource`/`Report` object would be providing inputs to that intermediate type.

Reply:
One controller driving another via a CRD can certainly make sense, but it can also make things more complicated and introduces another place for things to break.

In this case i'm torn between it being a nice separation of concerns, and it being an unnecessary decoupling that exposes users to an api they shouldn't be exposed to (makes your life harder if you wanted to migrate off postgrestable in the future, for example).

i'm guessing at this point it's simpler to keep it than get rid of it, so i won't stand in the way, but something to think about.